src/ray/core_worker/reference_count.cc

```diff
@@ -939,11 +939,16 @@ void ReferenceCounter::PopAndClearLocalBorrowers(
     // borrowed IDs. This is because we artificially increment each borrowed ID to
     // keep it pinned during task execution. However, this should not count towards
     // the final ref count / existence of local ref returned to the task's caller.
-    RAY_CHECK(GetAndClearLocalBorrowersInternal(borrowed_id,
-                                                /*for_ref_removed=*/false,
-                                                /*deduct_local_ref=*/true,
-                                                &borrowed_refs))
-        << borrowed_id;
+    bool ref_found = GetAndClearLocalBorrowersInternal(borrowed_id,
+                                                       /*for_ref_removed=*/false,
+                                                       /*deduct_local_ref=*/true,
+                                                       &borrowed_refs);
+    if (!ref_found) {
+      RAY_CHECK(ObjectID::IsActorID(borrowed_id))
```

Comment:
Should we make those "out-of-band" cases in-band instead so that reference counting is correct. With this PR, it's still possible that reference count table is 0 while the application still has a reference to the actor handle.